scipy/stats/_kde.py

```diff
@@ -574,17 +577,25 @@ def _compute_covariance(self):
         covariance_factor().
         """
         self.factor = self.covariance_factor()
-        # Cache covariance and inverse covariance of the data
-        if not hasattr(self, '_data_inv_cov'):
+        # Cache covariance and permuted cholesky decomp of permuted covariance
+        if not hasattr(self, '_data_cho_cov'):
             self._data_covariance = atleast_2d(cov(self.dataset, rowvar=1,
                                                bias=False,
                                                aweights=self.weights))
-            self._data_inv_cov = linalg.inv(self._data_covariance)
+            self._data_cho_cov = linalg.cholesky(
+                self._data_covariance[::-1, ::-1]).T[::-1, ::-1]
 
         self.covariance = self._data_covariance * self.factor**2
-        self.inv_cov = (self._data_inv_cov / self.factor**2).astype(np.float64)
-        L = linalg.cholesky(self.covariance*2*pi)
-        self.log_det = 2*np.log(np.diag(L)).sum()
+        self.cho_cov = (self._data_cho_cov * self.factor).astype(np.float64)
+        self.log_det = 2*np.log(np.diag(self.cho_cov
+                                        * np.sqrt(2*pi))).sum()
+
+    @property
+    def inv_cov(self):
+        self.factor = self.covariance_factor()
+        self._data_covariance = atleast_2d(cov(self.dataset, rowvar=1,
```

Comment:
Why do we recompute `self._data_covariance` here? Does it help to maintain backwards compatibility for existing subclasses? If this is needed, could  probably use a comment to explain why but it’s fine if you think it isn’t necessary.

Reply:
I think I assumed that since `_compute_covariance` used to re-calculate the covariance every time, there must have been a reason. Otherwise, why not do it just once in the `__init__` method? As it was, it was re-calculated every time `set_bandwidth` was called, so maybe people use `set_bandwidth` to recalculate everything after modifying the public attribute `dataset`? I don't really know, but figured it would be safer this way. 

And maybe subconsciously I want use of `inv_cov` to be as slow as possible : ) See the discussion in the original incarnation of this issue - https://github.com/scipy/scipy/pull/5087#issuecomment-775529643.

Reply:
Cool. That makes sense.